### PR TITLE
[com_media] Refactoring of thumbnails and details view

### PR DESCRIPTION
**Pull request by jissereitsma**
Part of the refactoring work of @Buddhima (https://github.com/joomla/joomla-cms/pull/3839) was to add better delete-actions and edit-actions for images. While working on updating his during a code sprint, we found that the current thumbnails and details view was lacking flexibility. This PR includes refactoring to allow for more flexible updating in future work.

This PR adds first of all JLayout files for delete() and edit() actions. Because an edit() action is currently missing (it is part of the bundled works of @Buddhima though), there is a simple check to see whether the "image" view is there or not. The delete() action has been refactored to use the Delete toolbar-button, instead of using its own link. This way, future checks in that Delete toolbar-button ("Hey, you did not select anything") also applies to these actions.

Because of these changes, the HTML has changed in various places as well: For instance, the icons have been made a little bit bigger. In future PRs, this code needs beautifying: Removing style-arguments with proper CSS rules, new CSS-rules for the thumbnail-sizes, etcetera.

For testing, apply the patch, and try to delete the various files in either Thumnails-view or Details-view: Use all types of files: Images, ZIP-files, PDF-files. Try to delete an empty folder as well. And try to delete a non-empty folder - because it should not work ;)

### CODE REVIEW

**Comment by wilsonge** [comment]
You appear to have added an empty file? (https://github.com/jissereitsma/joomla-cms/blob/new_com_media_jisse_01/administrator/components/com_media/controllers/image.php)

**Comment by jissereitsma** [comment]
@wilsonge Correct. During the code sprint, we started to implement a new MVC class-set for "image" as well. But later on, we decided to focus on some smaller tasks instead. I'll remove the file again.

**Comment by Kubik-Rubik** [comment]
@jissereitsma Could you please merge @zero-24's PR against your branch so that we make Travis happy? Thanks! :-)

**Comment by jissereitsma** [comment]
Many thanks, @zero-24 @Kubik-Rubik. The PR is merged in my branch.

File: administrator/components/com_media/views/medialist/tmpl/details.php [hunk 1]
```diff
@@ -25,41 +26,54 @@
...
+		<input type="hidden" name="boxchecked" value="0" />
+		<input type="hidden" name="task" value="list" />
+		<input type="hidden" name="username" value="" />
+		<input type="hidden" name="password" value="" />
+		<?php echo JHtml::_('form.token'); ?>
 	</div>
 </form>
+
+<script>
+	document.adminForm = document['mediamanager-form'];
```
Comment (wilsonge) [inline administrator/components/com_media/views/medialist/tmpl/details.php]: Please put this in the head using `JDocument::addScriptDeclaration` before @dgt41 gets the shotgun out :)

**Comment by peterlose** [comment]
I'm getting some notices:

skaermbillede 2015-08-25 kl 10 36 22

File: administrator/components/com_media/views/medialist/tmpl/details.php (hunk 1 above)
Comment (jissereitsma) [inline administrator/components/com_media/views/medialist/tmpl/details.php]: I ... don't ... know ... why ... this ... happened. Fixed.

**Comment by wilsonge** [comment]
@test I just had a play around with this and it seems like a good improvement to me :) The view in com_media was significantly better. No apparent issues using the image editor-xtd button either in frontend or backend editing.

My only slight issue was that I felt using the preview button wasn't 100% obvious as it was a relatively small icon. I would reintroduce clicking on the image also opening the preview to make things easy for users

**Comment by Worti2** [comment]
The Patch is not OK.
- In the thumbnail-view and the detail-view of the backend the delete-button next to create-new-folder-button is without function.
- If you select more than one picture in the thumbnail-view and if you click on the red x of one picture, you will delete all chosen (I do not know if this is an issue).
- In the detail-view it is not possible to choose all pictures with the check-all checkbox.
This is OK:
- You are not able to delete a folder if there are still pictures in.
Regards
Markus

**Comment by Worti2** [comment]
I have tested this item :red_circle: unsuccessfully on e6d05a6aec9814235e1811322b04beee37cdbafb<br /><br />The Patch is not OK.
- In the thumbnail-view and the detail-view of the backend the delete-button next to create-new-folder-button is without function.
- If you select more than one picture in the thumbnail-view and if you click on the red x of one picture, you will delete all chosen (I do not know if this is an issue).
- In the detail-view it is not possible to choose all pictures with the check-all checkbox.
This is OK:
- You are not able to delete a folder if there are still pictures in.
Regards Markus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7646">issues.joomla.org/joomla-cms/7646</a>.</sub>

**Comment by roland-d** [comment]
@jissereitsma Is this obsolete with the new work you are doing on the media manager?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7646">issues.joomla.org/joomla-cms/7646</a>.</sub>

**Comment by yireo** [comment]
Yes, it is. Can you close this PR?

**Comment by brianteeman** [comment]
Closed as requested<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7646">issues.joomla.org/joomla-cms/7646</a>.</sub>

**Pull request closed without merging**